hazelcast/src/main/java/com/hazelcast/jet/impl/connector/AbstractHazelcastConnectorSupplier.java

```diff
@@ -67,7 +73,23 @@ protected Processor createProcessor(HazelcastInstance instance, SerializationSer
 
     @Override
     public void init(@Nonnull Context context) {
-        if (clientXml != null) {
+        createHzClient(context);
+    }
+
+    private void createHzClient(ProcessorSupplier.Context context) {
+        // The order is important.
+        // If dataConnectionConfig is specified prefer it to clientXml
+        if (dataConnectionName != null) {
+            HazelcastDataConnection hazelcastDataConnection = context
+                    .dataConnectionService()
+                    .getAndRetainDataConnection(dataConnectionName, HazelcastDataConnection.class);
+            try {
+                instance =  hazelcastDataConnection.getClient();
+                serializationService = ((HazelcastClientProxy) instance).getSerializationService();
```

Comment:
I would not rely on technical details here; is there a reason why we cannot get serializationService in other way?